lib/main.dart

```diff
@@ -29,17 +29,62 @@ import 'package:pangolin/services/langpacks.dart';
 import 'package:pangolin/services/notifications.dart';
 import 'package:pangolin/services/power.dart';
 import 'package:pangolin/services/search.dart';
+import 'package:pangolin/services/shell.dart';
 import 'package:pangolin/services/tray.dart';
 import 'package:pangolin/services/wm.dart';
+import 'package:stack_trace/stack_trace.dart';
 import 'package:yatl_flutter/yatl_flutter.dart';
 import 'package:zenit_ui/zenit_ui.dart';
 
+void _pushShellNotification(LogRecord record) {
```

Comment:
Consider moving this function as well as the Logger listener into it's own class/file